torch/distributed/device_mesh.py

```diff
@@ -392,81 +95,123 @@ def _get_device_handle(device_type: str = "cuda"):
         """
         return getattr(torch, device_type, None)
 
-    class DeviceMesh:
-        """
-        DeviceMesh represents a mesh of devices, where layout of devices could be
-        represented as a n-d dimension array, and each value of the n-d dimensional
-        array is the global id of the default process group ranks.
+    class Layout:
```

Comment:
I guess this is mainly the PyTorch API contract issue: we would need to bring all the classes/functions defined to private namespace, unless we want to make them public to users.

Reply:
Yes it should be enough! iirc all class/functions lives in a public namespace (like device_mesh.py) would be exposed as public APIs, unless explicitly prepend an underscore.